gammapy/cube/fit.py

```diff
@@ -267,6 +268,63 @@ def npred(self):
 
         return npred_total
 
+    @classmethod
+    def create(cls, geom, geom_irf=None, migra_axis=None, rad_axis=None):
+        """Creates a MapDataset object with zero filled maps
+
+        Parameters
+        ----------
+        geom: `~gammapy.maps.WcsGeom`
+            Reference target geometry in reco energy, used for counts and background maps
+        geom_irf: `~gammapy.maps.WcsGeom`
+            Reference image geometry in true energy, used for IRF maps.
+        migra_axis: `~gammapy.maps.MapAxis`
+            Migration axis for the energy dispersion map
+        rad_axis: `~gammapy.maps.MapAxis`
+            Rad axis for the psf map
+        """
+
+        geom_irf = geom_irf or geom
+
+        counts = Map.from_geom(geom, unit="")
+
+        background = Map.from_geom(geom, unit="")
+        background_model = BackgroundModel(background)
+
+        energy_axis = geom_irf.get_axis_by_name("ENERGY")
+
+        exposure_geom = geom.to_image().to_cube([energy_axis])
+        exposure = Map.from_geom(exposure_geom, unit="m2 s")
+
+        mask = np.ones(geom.data_shape, dtype=bool)
+
+        gti = GTI.create([0 * u.s], [0 * u.s])
+
+        edisp, psf = None, None
```

Comment:
If we want the default binning to be the binnings in the IRFs (as implemented in #2358), then its a bit difficult to do it inside `MapDataset` as the `MapDataset` as no idea about the observations and the IRFs being used to create the same. So then the binning here has to be completely arbitrary.
That is why I thought maybe it would be better to have a small helper function to create default binnings, and this is what is passed to the `MapDataset.create()` normally.

Reply:
OK, I see. In https://github.com/gammapy/gammapy/blob/master/gammapy/cube/make.py#L332 you also copy the binning from the `PSF3D` and `EnergyDispersion2D` objects. Actually I think it might still be a good idea to introduce somewhere the default `RAD_AXIS` and `MIGRA_AXIS` objects, they could be used in both places then.

Reply:
We can put some default value, but it is clear that for a regular (high level driven) analysis the geometry will be defined on initialization of the loop. Even if default values are entered in the config file, we should probably never use default values in this `create` methods but pass a default geometry deduced from IRF at the start. 
The function in charge to create this default geometry has to know about the IRF used in the datastore, so this cannot be the job of `Dataset.create`

Reply:
In my view, the `.create()` method should make it simple to create an empty `MapDataset`, without much configuration, just like we have for `Map.create()`. I think in this case it's even desired, that we make a choice on good default values. For this method I would also choose the most intuitive parametrisation (e.g. `MapDataset.create(geom, energy_axis_true=None, rad_axis=None, migra_axis=None, bins_irf="0.2 deg", margin_irf="0.5 deg")`. Most of the users will never specify `rad_axis` nor `migra_axis` or `bins_irf`...

I can see the point of copying the binning from the exported IRFs, but on the other hand I think it's not even consistent between e.g. HAP and HAP-France exported data. Also I think that the choice of the exported binning was rather arbitrary and not optimised in terms of memory vs. precision.

In addition we could maybe add `MapDataset.from_geoms(geom, geom_true, geom_psf, geom_edisp)`, which is more flexible and offers full control. `.create()` would internally call `.from_geoms()`.